src/realm/error_codes.cpp

```diff
@@ -187,7 +251,68 @@ static const std::map<std::string_view, ErrorCodes::Error> error_codes_map = {
     {"InvalidReadOnlySchemaChange", ErrorCodes::InvalidReadOnlySchemaChange},
     {"InvalidExternalSchemaChange", ErrorCodes::InvalidExternalSchemaChange},
     {"DeleteOnOpenRealm", ErrorCodes::DeleteOnOpenRealm},
-    {"NotSupported", ErrorCodes::NotSupported}};
+    {"NotSupported", ErrorCodes::NotSupported},
+    {"ClientUserNotFound", ErrorCodes::ClientUserNotFound},
+    {"ClientUserNotLogged_in", ErrorCodes::ClientUserNotLoggedIn},
+    {"ClientAppDeallocated", ErrorCodes::ClientAppDeallocated},
+    {"BadToken", ErrorCodes::BadToken},
+    {"MalformedJson", ErrorCodes::MalformedJson},
+    {"MissingJsonKey", ErrorCodes::MissingJsonKey},
+    {"BadBsonParse", ErrorCodes::BadBsonParse},
+
+    {"CustomError", ErrorCodes::CustomError},
+    {"MissingAuthReq", ErrorCodes::MissingAuthReq},
+    {"InvalidSession", ErrorCodes::InvalidSession},
+    {"UserAppDomainMismatch", ErrorCodes::UserAppDomainMismatch},
+    {"DomainNotAllowed", ErrorCodes::DomainNotAllowed},
+    {"ReadSizeLimitExceeded", ErrorCodes::ReadSizeLimitExceeded},
+    {"InvalidParameter", ErrorCodes::InvalidParameter},
+    {"MissingParameter", ErrorCodes::MissingParameter},
+    {"TwilioError", ErrorCodes::TwilioError},
+    {"GCMError", ErrorCodes::GCMError},
+    {"HTTPError", ErrorCodes::HTTPError},
+    {"AWSError", ErrorCodes::AWSError},
+    {"MongoDBError", ErrorCodes::MongoDBError},
+    {"ArgumentsNotAllowed", ErrorCodes::ArgumentsNotAllowed},
+    {"FunctionExecutionError", ErrorCodes::FunctionExecutionError},
+    {"NoMatchingRule", ErrorCodes::NoMatchingRule},
+    {"InternalServerError", ErrorCodes::InternalServerError},
+    {"AuthProviderNotFound", ErrorCodes::AuthProviderNotFound},
+    {"AuthProviderAlreadyExists", ErrorCodes::AuthProviderAlreadyExists},
+    {"ServiceNotFound", ErrorCodes::ServiceNotFound},
+    {"ServiceTypeNotFound", ErrorCodes::ServiceTypeNotFound},
+    {"ServiceAlreadyExists", ErrorCodes::ServiceAlreadyExists},
+    {"Service_commandNotFound", ErrorCodes::Service_commandNotFound},
+    {"ValueNotFound", ErrorCodes::ValueNotFound},
+    {"ValueAlreadyExists", ErrorCodes::ValueAlreadyExists},
+    {"Value_duplicateName", ErrorCodes::Value_duplicateName},
```

Comment:
There are a few strings that have underscores in them now and didn't get copied over correctly? Due to the nature of how we use `ErrorCodes::from_string()` to match the string that the server sends us into an error code, these typos are going to break that translation step. Perhaps consider adding a comment on this section telling us not to change these string values in the future either.

Reply:
I thought I got rid of all '_'. Perhaps I should have searched for them :-)